developer_manual/client_apis/OCS/ocs-text2image-api.rst

```diff
@@ -0,0 +1,176 @@
+* Response:
+    - Status code:
+        + ``200 OK``
+    - Data:
+
++----------------------+--------+---------------------------------------------------------------------------------------------------------------+
+| field                | type   | Description                                                                                                   |
++----------------------+--------+---------------------------------------------------------------------------------------------------------------+
+|``isAvailable``       | bool   | Boolean indicating whether any Text-To-Image providers are installed                                          |
++----------------------+--------+---------------------------------------------------------------------------------------------------------------+
+
+Schedule a task
+---------------
+
+.. versionadded:: 28
+
+.. note:: The endpoint is rate limited as it can be quite resource intensive. Users can make 20 requests in 2 minutes, guests only 5
+
+* Method: ``POST``
+* Endpoint: ``/schedule``
+* Data:
+
++-------------------+-------------+--------------------------------------------------------------------------------+
+| field             | type        | Description                                                                    |
++-------------------+-------------+--------------------------------------------------------------------------------+
+|``input``          | string      | The input text for the task                                                    |
++-------------------+-------------+--------------------------------------------------------------------------------+
+|``numberOfImages`` | int         | The number of images to generate (optional; default: 8)                        |
++-------------------+-------------+--------------------------------------------------------------------------------+
+|``appId``          | string      | The id of the requesting app                                                   |
++-------------------+-------------+--------------------------------------------------------------------------------+
+|``identifier``     | string      | An app-defined identifier for the task (optional)                              |
++-------------------+-------------+--------------------------------------------------------------------------------+
+
+If possible the task will be executed while the request is processed on the server, otherwise it is scheduled as a background job.
+
+* Response:
+    - Status code:
+        + ``200 OK``
+        + ``412 Precondition Failed`` - When the task type is not available currently
+        + ``429 Too Many Requests`` - When the rate limiting was exceeded
+
+    - Data:
+        + ``id`` - Only provided in case of ``200 OK``, the assigned task id, int
+        + ``input`` - Only provided in case of ``200 OK``, the task input, string
+        + ``status`` - Only provided in case of ``200 OK``, the current task status, int, see backend API
```

Comment:
I think you've added a lot of useful cross refs. Adding cross refs to the correct subsections for the occurences of "see backend API" in this file could save some time when figuring out how to handle the endpoint outputs...?

Reply:
Good idea!